### DISCUSSION TITLE
message select support for unordered messages

### DISCUSSION THREAD

**Discussion by hunjixin** [filecoin-project/lotus#6213] [Ideas]
in current implement, if two miner package message of the same address, if blockA has nonce 1,2,3 and blockB has nonce 4,5. the result of process ,maybe 1,2,3,4,5 or 1,2,3. this depends on the ticket value and cid.  if blockA is before blockB we can get 1,2,3,4,5. but if blockB before blockA,  message with nonce4,5 will be ignore, only process 1,2,3. so why not sort msg in tipset to avoid this problem.    

in venus,  when two miner has block in the same tipset, they cannot select different messager of the same address due to uncertain cid valye

**Comment by de-authority** [comment]
Hey. Although this information is a bit out of date, does this problem still exist?

**Reply by LexLuthr** [reply]
@hunjixin Are you still facing this problem?
